deployment/paiLibrary/paiOrchestration/change_node.py

```diff
@@ -16,36 +16,49 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 import os
+import yaml
 import logging
 
 import temp_config
 import temp_kubespray
 from ..common import linux_shell
+from ...confStorage.upload import UploadConfiguration
 
 
-class AddNode:
+class ChangeNode:
 
-    def __init__(self, kube_config_path=None, node_list=None, verbose=False):
+    def __init__(self, kube_config_path=None, verbose=False):
         self._logger = logging.getLogger(__name__)
         self._kube_config_path = kube_config_path
-        self._node_list = node_list
         if verbose:
             self._ansible_callback_vars = "export ANSIBLE_DISPLAY_OK_HOSTS=yes && export ANSIBLE_DISPLAY_SKIPPED_HOSTS=yes && export ANSIBLE_CALLBACK_WHITELIST=\"profile_tasks\" &&"
         else:
             self._ansible_callback_vars = "export ANSIBLE_DISPLAY_OK_HOSTS=no && export ANSIBLE_DISPLAY_SKIPPED_HOSTS=no && export ANSIBLE_CALLBACK_WHITELIST=\"\" &&"
 
-    def run(self):
+    def _update_layout_yaml(self, layout_yaml_path, remove_node_list):
+        self._logger.info("Remove nodes in `layout.yaml`")
+        with open(layout_yaml_path, "r") as f:
+            layout_data = yaml.load(f, yaml.SafeLoader)
+        layout_data["machine-list"] = [host for host in layout_data["machine-list"] if host["hostname"] not in remove_node_list]
+        with open(layout_yaml_path, "w") as f:
+            yaml.dump(layout_data, f, default_flow_style=False)
+
+    def run(self, mode="add", node_list=[]):
         temp_kubespray_folder = temp_kubespray.TempKubespray()
         temp_config_folder = temp_config.TempConfig(self._kube_config_path)
-        node_list_string = ",".join(self._node_list)
-        self._logger.info("Begin to add nodes: {}".format(node_list_string))
+        node_list_string = ",".join(node_list)
+        self._logger.info("Begin to {} nodes: {}".format(mode, node_list_string))
         linux_shell.execute_shell_raise(
-            shell_cmd="cd {} && {} ansible-playbook -i {} scale.yml -b --become-user=root -e \"@{}\" -e \"node={}\"".format(
+            shell_cmd="cd {} && {} ansible-playbook -i {} {} -b --become-user=root -e \"@{}\" -e \"node={}\"".format(
```

Comment:
The nodes to add are specified with the `--limit` flag instead of `-e` when add nodes, https://github.com/microsoft/pai/blob/master/docs/manual/cluster-admin/how-to-add-and-remove-nodes.md#add-the-nodes-into-kubernetes